private/protocol/eventstream/eventstreamapi/signer.go

```diff
@@ -0,0 +1,43 @@
+package eventstreamapi
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/aws/aws-sdk-go/private/protocol/eventstream"
+)
+
+const (
+	chunkSignatureHeader = ":chunk-signature"
+	chunkDateHeader      = ":date"
+)
+
+// EventStreamSigner defines an interface for the implementation of signing of event stream payloads
+type EventStreamSigner interface {
+	GetSignature(headers, payload []byte, date time.Time) ([]byte, error)
+}
```

Comment:
Minor preference, but what do you think about naming this, `StreamSigner`? I think the EventStream StreamSigner, and aws-chunk StreamSigner are the same underlying behavior.

Reply:
Working on getting clarification on this out-of-band.